js/fibonacci.js

```diff
@@ -0,0 +1,155 @@
+// constants
+
+const FIB_URL = "http://localhost:5050/fibonacci/";
+const HIST_URL = "http://localhost:5050/getFibonacciResults";
+
+// page elements
+
+const button = document.getElementById("submit");
+const fibResult = document.querySelector(".fibonacci");
+const input = document.getElementById("inputNumber");
+const spinner = document.querySelector(".spinner");
+const mainSpinner = document.querySelector(".main-spinner");
+const history = document.getElementById("search-history");
+const check = document.getElementById("saveCalc");
+const dropdown = document.querySelector(".dropdown-menu");
```

Comment:
To be honest, I have no real idea why the inconsistency. I think that in the future, to select specific elements I will try to default to ids, unless I need to use querySelectorAll, for the sake of consistency.

It might have to do with the fact that in a basic course I did on Frontend Masters (with Brian Holt), he was talking about specificity in CSS and how ids are more specific than classes and we should default to classes, and it sort of made me reluctant overall to add ids to elements? Of course, styling is one thing and grabbing page elements for the JS is another.

Reply:
I would actually recommend you to use querySelector and querySelectorAll for everything, but up to you

Reply:
If querySelector is best practice then I'll do that! Thank you.